src/Common/AzurePSCmdlet.cs

```diff
@@ -110,16 +110,18 @@ protected string PSVersion
             }
         }
 
+        private string _moduleName;
+        private string _moduleVersion;
         /// <summary>
         /// Gets the PowerShell module name used for user agent header.
         /// By default uses "Azure PowerShell"
         /// </summary>
-        protected virtual string ModuleName { get { return "AzurePowershell"; } }
+        protected virtual string ModuleName { get { return _moduleName; } set { this._moduleName = value; } }
```

Comment:
If there's no special logic in getter and setter, this is equivalant to
```suggestion
        protected virtual string ModuleName { get; set; }
```